### PR TITLE
Submission of DSA Assignment 1

### PR DESCRIPTION
**Pull request by meta-chinmay-jain**
- LinkedList class implements the linked list functionalities.
- Main class provides the driver code.

### CODE REVIEW

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	 */
+	public boolean rotateSubList(int leftPos, int rightPos, int noOfShifts) {
+		if (leftPos < 1 || leftPos > rightPos || rightPos > length || rightPos < leftPos) {
+			return false;
+		}
+		Node temp = head;
+		Node ptr1 = null;
+		Node ptr2;
+		Node ptr3;
+		int i;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:83]: Done.

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+/**
+ * This class defines methods to create, display and rotate a Linked List.
+ * 
+ * @author Chinmay Jain
+ * @version 1.0
+ * @since 2021-03-04
+ */
+public class LinkedList {
+	Node head = null;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:11]: Done

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	int length = 0;
+
+	/**
+	 * This class is used to define the structure of a single Node of Linked List.
+	 */
+	public class Node {
+		Node next;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:18]: Done

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	 * @param leftPos    Start Position of Sub-List from List.
+	 * @param rightPos   End Position of Sub-List from List.
+	 * @param noOfShifts Number of shifts to be performed.
+	 * @return boolean True if rotation successful, false otherwise.
+	 */
+	public boolean rotateSubList(int leftPos, int rightPos, int noOfShifts) {
+		if (leftPos < 1 || leftPos > rightPos || rightPos > length || rightPos < leftPos) {
+			return false;
+		}
+		Node temp = head;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:79]: Done

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+		int i;
+
+		i = 1;
+		while (i < leftPos) {
+			ptr1 = temp;
+			temp = temp.next;
+			i++;
+		}
+
+		int subListLength = rightPos - leftPos + 1;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:92]: Done

File: Data Structures and Algorithms - GET/Assignment 1/src/com/poly/Polynomial.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+/**
+ * This class contains the driver code for Multivariate Polynomial
+ * implementation and stores a Polynomial representation into nested lists.
+ * 
+ * @author Chinmay Jain
+ * @version 1.0
+ * @since 2021-03-04
+ */
+public class Polynomial {
+	private static Scanner input = new Scanner(System.in);
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/poly/Polynomial.java:17]: Understood what needs to be done if such situation is to be handled.

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 7]
```diff
@@ -0,0 +1,144 @@
+package com.list;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:1]: Resolved issue.
Done.

File: Data Structures and Algorithms - GET/Assignment 1/src/com/poly/Polynomial.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/poly/Polynomial.java:17]: Handled issue with 0 as coefficient, as well as + sign in the beginning will not appear now.

File: Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+		i = 1;
+		while (i < subListLength - noOfShifts) {
+			temp = temp.next;
+			i++;
+		}
+		ptr2 = temp;
+
+		i = 1;
+		while (i <= noOfShifts) {
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/list/LinkedList.java:106]: Added algo in method spec.
Done.

File: Data Structures and Algorithms - GET/Assignment 1/src/com/poly/Polynomial.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Data Structures and Algorithms - GET/Assignment 1/src/com/poly/Polynomial.java:17]: Done

**Review by meta-love-gupta — APPROVED** [review]

**Pull request merged**